test/e2e/metrics_e2e_test.go

```diff
@@ -5,59 +5,83 @@ package e2e
 import (
 	"context"
 	"fmt"
-	"regexp"
-
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
-	log "github.com/sirupsen/logrus"
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/api/client/clientset/versioned"
+	"github.com/operator-framework/operator-lifecycle-manager/pkg/lib/operatorclient"
+	"github.com/operator-framework/operator-lifecycle-manager/test/e2e/ctx"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/net"
-
-	"github.com/operator-framework/api/pkg/operators/v1alpha1"
-	"github.com/operator-framework/operator-lifecycle-manager/pkg/lib/operatorclient"
+	"regexp"
 )
 
-var _ = Describe("Metrics", func() {
-	It("endpoint", func() {
-
-		// TestMetrics tests the metrics endpoint of the OLM pod.
-
-		c := newKubeClient(GinkgoT())
-		crc := newCRClient(GinkgoT())
-
-		failingCSV := v1alpha1.ClusterServiceVersion{
-			TypeMeta: metav1.TypeMeta{
-				Kind:       v1alpha1.ClusterServiceVersionKind,
-				APIVersion: v1alpha1.ClusterServiceVersionAPIVersion,
-			},
-			ObjectMeta: metav1.ObjectMeta{
-				Name: genName("failing-csv-test-"),
-			},
-			Spec: v1alpha1.ClusterServiceVersionSpec{
-				InstallStrategy: v1alpha1.NamedInstallStrategy{
-					StrategyName: v1alpha1.InstallStrategyNameDeployment,
-					StrategySpec: strategy,
-				},
-			},
-		}
-
-		cleanupCSV, err := createCSV(GinkgoT(), c, crc, failingCSV, testNamespace, false, false)
-		Expect(err).ToNot(HaveOccurred())
-		defer cleanupCSV()
+var _ = Describe("Metrics are generated for OLM pod", func() {
 
-		_, err = fetchCSV(GinkgoT(), crc, failingCSV.Name, testNamespace, csvFailedChecker)
-		Expect(err).ToNot(HaveOccurred())
+	var (
+		c   operatorclient.ClientInterface
+		crc versioned.Interface
+	)
+
+	BeforeEach(func() {
+		c = newKubeClient(GinkgoT())
+		crc = newCRClient(GinkgoT())
+
+	})
 
-		// Verify metrics have been emitted for packageserver csv
-		Expect(getMetricsFromPod(c, getOLMPod(c), "8081")).To(And(
-			ContainSubstring("csv_abnormal"),
-			ContainSubstring(fmt.Sprintf("name=\"%s\"", failingCSV.Name)),
-			ContainSubstring("phase=\"Failed\""),
-			ContainSubstring("reason=\"UnsupportedOperatorGroup\""),
-			ContainSubstring("version=\"0.0.0\""),
-			ContainSubstring("csv_succeeded"),
-		))
+	Context("Given an operator group that supports all namespaces", func() {
```

Comment:
Nit: It would be nice to call out the dependency (maybe with By) on the pre-existing `testNamespace` and on the default OperatorGroup potentially set up in BeforeSuite. It's Monday morning, so it took me a moment to realize why there was no test fixture setup in this context.